File: Technology Transfer _ Sulambi VMS/Source Code/sulambi-frontend-main/sulambi-frontend-main/src/api/membership.ts
import axios from "./init";

const basePath = "/membership";

export const getAllMembers = () => {
  return axios.get(`${basePath}/`);
};

export const approveMembership = (id: number) => {
  return axios.patch(`${basePath}/approve/${id}`);
};

export const rejectMembership = (id: number) => {
  return axios.patch(`${basePath}/reject/${id}`);
};

export const activateMember = (id: number) => {
  return axios.patch(`${basePath}/activate/${id}`);
};

export const deactivateMember = (id: number) => {
  return axios.patch(`${basePath}/deactivate/${id}`);
};

File: Technology Transfer _ Sulambi VMS/Source Code/sulambi-backend-main/sulambi-backend-main/app/routes/membership.py
from flask import Blueprint
from ..controllers import membership
from ..middlewares import tokenCheck

MembershipBlueprint = Blueprint('membership', __name__, url_prefix="/membership")

@MembershipBlueprint.get("/")
def getAllMembershipRoute():
  return membership.getAllMembership()

@MembershipBlueprint.patch("/approve/<membershipRequestId>")
def approveMembership(membershipRequestId):
  return membership.approveMembership(membershipRequestId)

@MembershipBlueprint.patch("/reject/<membershipRequestId>")
def rejectMembership(membershipRequestId):
  return membership.rejectMembership(membershipRequestId)

@MembershipBlueprint.patch("/activate/<membershipRequestId>")
def activateMembership(membershipRequestId):
  return membership.activateMembership(membershipRequestId)

@MembershipBlueprint.patch("/deactivate/<membershipRequestId>")
def deactivateMembership(membershipRequestId):
  return membership.deactivateMembership(membershipRequestId)
